### PR TITLE
Add RDS Postgres end-to-end tests

### PR DESCRIPTION
**Pull request by GavinFrazar**
This PR adds end-to-end tests for AWS RDS Postgres instances.
I limited it to just Postgres instance for this PR, for smaller scope and easier review.

It tests that:
- The Teleport Discovery Service discovers RDS Postgres databases and creates them in Teleport
- The Teleport Database Service can connect a user to an RDS Postgres instance via IAM auth as db user `teleport-ci-e2e-test`
  - the database user will have already been provisioned by terraform with IAM auth enabled.

GHA will assume the intermediate role `arn:aws:iam::307493967395:role/tf-aws-e2e-gha-role` as it does for the existing kube e2e tests.

Then the role `arn:aws:iam::307493967395:role/ci-databases-database-svc` can be assumed for database access (connecting users)
and the role `arn:aws:iam::307493967395:role/ci-databases-discovery-svc` can be assumed for discovering databases.

The test can be run manually by deploying the required infra in AWS and setting the appropriate env vars, for example:
```bash
$ export AWS_REGION=us-west-1
$ export DATABASE_USER=teleport-ci-e2e-test
$ export DATABASE_SERVICE_ASSUME_ROLE=arn:aws:iam::278576220453:role/ci-database-e2e-tests-database-svc
$ export DATABASE_DISCOVERY_SERVICE_ASSUME_ROLE=arn:aws:iam::278576220453:role/ci-database-e2e-tests-discovery-svc
$ export DISCOVERY_MATCHER_LABELS="env=ci"
$ export RDS_POSTGRES_INSTANCE_NAME=ci-database-e2e-tests-rds-postgres-instance-us-west-1-278576220453
$ make e2e-aws ADDFLAGS='-run TestDatabases' 
```
Part one of implementing https://github.com/gravitational/teleport/pull/27156 for databases.

TODO Followup PRs: 
- [x] RDS MySQL instance tests

issue: https://github.com/gravitational/teleport/issues/27325

### CODE REVIEW

File: e2e/aws/rds_test.go
Comment (tigrato) [inline e2e/aws/rds_test.go:206]: For something that runs each 1min, testing the tick each 500ms seems to much
```suggestion
	}, 3*time.Minute, 3*time.Second, "waiting for the discovery service to create databases")
```

File: e2e/aws/rds_test.go
Comment (tigrato) [inline e2e/aws/rds_test.go:45]: Can you include a test scenario where the database doesn't match the db label and so it's not discovered

File: e2e/aws/rds_test.go
Comment (tigrato) [inline e2e/aws/rds_test.go:111]: Can't you reuse the logic from https://github.com/gravitational/teleport/blob/1178971b3d7c6f82d93c476b45c1c9ebd3b5487f/integration/db/db_integration_test.go#L194-L207

File: e2e/aws/rds_test.go
Comment (GavinFrazar) [reply e2e/aws/rds_test.go:124]: it was enough when I tried it a few times in the dev account - I tried that by deleting the inline policy that the db service auto-adds to itself and re-running.

we can extend it if we see failures, although typically the first connection will succeed since the inline policy from the first run will persist.

File: e2e/aws/rds_test.go
Comment (tigrato) [inline e2e/aws/rds_test.go:45]: Can you skip the test by default similarly to Kube?
if you need to debug something local and the number of tests is large, it's cool to skip them by default instead of having to check what's running

File: e2e/aws/rds_test.go
Comment (GavinFrazar) [reply e2e/aws/rds_test.go:111]: I can add a test for direct connection to the cluster proxy reusing that func, but still would keep what I have to test the local proxy path

File: e2e/aws/rds_test.go
Comment (GavinFrazar) [reply e2e/aws/rds_test.go:45]: I can, although I didn't want other CI workflows to be picking up the e2e test suite and needlessly building just to skip the tests, so I figured it would be useful to fail fast

File: e2e/aws/rds_test.go
Comment (GavinFrazar) [reply e2e/aws/rds_test.go]: [093bec5](https://github.com/gravitational/teleport/pull/29755/commits/093bec5940eac9a36dcbbf98ae2344809e87bc93)

File: e2e/aws/rds_test.go
Comment (GavinFrazar) [reply e2e/aws/rds_test.go:212]: added direct conn test as well [ddce196](https://github.com/gravitational/teleport/pull/29755/commits/ddce19696d334e59a3e51e7d09199f06543868b2)

File: e2e/aws/rds_test.go
Comment (GavinFrazar) [reply e2e/aws/rds_test.go:50]: [245b6a6](https://github.com/gravitational/teleport/pull/29755/commits/245b6a6a238eeed9e4a8656c6ed2009e157ea6ba)

Comment (GavinFrazar) [reply e2e/aws/rds_test.go:50]: added in https://github.com/gravitational/teleport/pull/29755/commits/245b6a6a238eeed9e4a8656c6ed2009e157ea6ba

Comment (GavinFrazar) [reply e2e/aws/rds_test.go:50]: and afa6118490da717479fbc6f01ea7c275dd4e594b

File: e2e/aws/eks_test.go
Comment (GavinFrazar) [reply e2e/aws/eks_test.go]: 632152ccb2b43c64b5ad5a314d24ac3e2b1e3197

File: e2e/aws/rds_test.go
Comment (greedy52) [reply e2e/aws/rds_test.go:225]: > we can extend it if we see failures, although typically the first connection will succeed since the inline policy from the first run will persist.

Should we clean up the policy before each test? If we are not testing the db/cloud/iam service, might just give the `rds:connect` permissions for the specific RDS resource to the agent, instead of giving it `iam:PutRolePolicy`

Comment (GavinFrazar) [reply e2e/aws/rds_test.go:225]: I prefer keeping things closer to real setup, even though it makes the test slows by a few seconds

**Review by greedy52 — APPROVED** [review]

File: e2e/aws/main_test.go
Comment (greedy52) [reply e2e/aws/main_test.go:97]: I am not familiar with the setup, are we able to add that CA to the system? That would be great.

I was thinking to programmatically providing a cert pool with the self signed certs to all places needed.

Comment (tigrato) [reply e2e/aws/main_test.go:97]: Yes, before running the tests our workflow should:
- download mkcert or embed it into our build image
- `mkcert -install`
- `mkcert localhost "*.localhost" localhost 127.0.0.1 ::1`

At this point, our system already trusts the mkcert generate CA and we need to configure the cert/key pair under `proxy_service.https_keypairs`

**Review by tigrato — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Comment by GavinFrazar** [comment]
@zmb3 @r0mant needs code owner approval for the workflow changes

**Review by r0mant — APPROVED** [review]
Bot.

**Pull request merged**
